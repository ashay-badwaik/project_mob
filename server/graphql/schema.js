const { gql } = require('apollo-server');


const typeDefs = gql`
    type UserDetails{
        id: ID!
        name: String
        bio: String
        image_path: String
    }


    type Query {
        getAllUserDetails: [UserDetails]!
        getUserDetails(id:ID!): UserDetails!
    }

    type Mutation {
        addUser(name:String! bio:String!): String!
        uploadImage(id:ID! path:String!): String!
        uploadPath(id:ID! path:String!): String!
    }
`

module.exports = typeDefs;