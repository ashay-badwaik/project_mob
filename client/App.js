/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';

import { Text, View } from 'react-native';

import HomeScreen from './components/homeScreen';
import StoryScreen from './components/storyScreen';
import AddStoryScreen from './components/addStoryScreen';
import ChangeProfilePic from './components/changeProfilePic';


const client = new ApolloClient({
  uri: 'http://localhost:4000/',
  cache: new InMemoryCache(),
})

const Stack = createNativeStackNavigator();

const App = () => {
  return (
    <ApolloProvider client={client}>
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Home">
          <Stack.Screen name="Home" component={HomeScreen} options={{headerShown: false}}/>
          <Stack.Screen name="Story" component={StoryScreen}/>
          <Stack.Screen name="AddStory" component={AddStoryScreen} />
          <Stack.Screen name="ChangeProfilePic" component={ChangeProfilePic} />
        </Stack.Navigator>
      </NavigationContainer>
    </ApolloProvider>
  );
}

export default App;project_mov
