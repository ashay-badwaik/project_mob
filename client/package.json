{
  "name": "client",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@apollo/client": "^3.4.7",
    "@react-navigation/native": "^6.0.1",
    "@react-navigation/native-stack": "^6.0.2",
    "graphql": "^15.5.1",
    "react": "17.0.1",
    "react-native": "^0.64.2",
    "react-native-image-crop-picker": "^0.36.2",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-screens": "^3.5.0",
    "react-native-svg": "^9.13.6",
    "react-native-svg-transformer": "^0.14.3"
  },
  "devDependencies": {
    "@babel/core": "7.15.0",
    "@babel/runtime": "7.14.8",
    "@react-native-community/eslint-config": "2.0.0",
    "babel-jest": "26.6.3",
    "eslint": "7.14.0",
    "jest": "26.6.3",
    "metro-react-native-babel-preset": "0.64.0",
    "react-test-renderer": "17.0.1"
  },
  "jest": {
    "preset": "react-native"
  }
}
