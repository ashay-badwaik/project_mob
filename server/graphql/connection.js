const mysql = require('mysql2');

const connection = mysql.createConnection({
    host: 'localhost',
    port: 3306, 
    user: 'root',
    password: '@vengers',
    database: 'story'
});

module.exports = connection;