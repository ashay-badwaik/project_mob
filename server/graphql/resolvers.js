const mysql = require('mysql2');
const fs = require("fs");
const connection = require("./connection");


const Query = {
    getAllUserDetails: async () => {
        let QUERY = `
            SELECT id, name, bio, image_path 
            FROM users
        `
        // console.log(QUERY);
        try {
            let data = await connection.promise().query(QUERY);
            // console.log(data[0]);
            return data[0];
        } catch (err) {
            console.log(err);
        }
    },

    getUserDetails: async (_, { id }) => {
        let QUERY = `
            SELECT id, name, bio, image_path 
            FROM users 
            WHERE id=${id}
        `
        try {
            let data = await connection.promise().query(QUERY);
            // console.log(data[0][0]);
            return data[0][0];
        } catch (err) {
            console.log(err);
        }
    }
}


const Mutation = {
    addUser: async (_, { name, bio }) => {
        let QUERY = `
            INSERT INTO users (name, bio) 
            VALUES ("${name}", "${bio}")
        `

        try {
            const success = await connection.promise().query(QUERY);
            if (success) return "Success";
        } catch (err) {
            console.log(err);
        }
    },

    uploadImage: async (_, { id, path }) => {
        let QUERY = `
            UPDATE trial 
            SET picture=LOAD_FILE('${path}') 
            WHERE id=${id}
        `

        try {
            const success = await connection.promise().query(QUERY);
            if (success) return "image uploaded";
        } catch (err) {
            console.log(err);
        }
    },

    uploadPath: async (_, { id, path }) => {
        let QUERY = `
            UPDATE users 
            SET image_path='${path}' 
            WHERE id=${id}
        `

        try {
            const success = await connection.promise().query(QUERY);
            if (success) return "image path uploaded";
        } catch (err) {
            console.log(err);
        }
    }
}

module.exports = { Query, Mutation }; 