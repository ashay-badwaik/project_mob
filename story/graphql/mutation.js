import { gql } from "@apollo/client";


export const UPLOAD_IMAGE_PATH = gql`
    mutation uploadPath($id: ID!, $path: String!) {
        uploadPath(
            id: $id 
            path: $path
        )
    }
`