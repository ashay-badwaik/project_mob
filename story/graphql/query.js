import { gql } from '@apollo/client';

export const USER_DETAILS = gql`
  query getUserDetails($id: ID!){
    getUserDetails(id:$id){
      name
      bio
      imagePath
    }
  }
`