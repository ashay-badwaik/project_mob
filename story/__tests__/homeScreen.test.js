import 'react-native';
import React from 'react';
import { MockedProvider } from "@apollo/client/testing";
import HomeScreen from '../components/homeScreen';
import { USER_DETAILS } from './../graphql/query';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import wait from "waait"
import { GraphQLError } from 'graphql';


const mockedParams = {
	route: {
		params: {
			available: true,
			viewed: true,
		}
	},
	navigation: {
		navigate: jest.fn()
	}
}

const tree = renderer.create(
	<MockedProvider>
		<HomeScreen {...mockedParams} />
	</MockedProvider>
)

it("snapshot", () => {
	expect(tree).toMatchSnapshot();
});

it("text feilds", () => {
	expect(tree.root.findAllByType("Text").length).toBe(3);
})

describe('navigation', () => {
	it("navigate to add story screen", () => {
		const btn = tree.root.findByProps({ testID: 'addBtn' }).props;
		btn.onPress();

		expect(mockedParams.navigation.navigate).toHaveBeenCalledWith(expect.objectContaining({ name: 'AddStory' }));
	});

	it("navigate to story screen", () => {
		const btn = tree.root.findByProps({ testID: 'image' }).props;
		btn.onPress();

		expect(mockedParams.navigation.navigate).toBeCalledWith("Story");
	});

	it('navigate to change profile pic screen', () => {
		const btn = tree.root.findByProps({ testID: 'image' }).props;
		btn.onLongPress();

		expect(mockedParams.navigation.navigate).toBeCalledWith(expect.objectContaining({ name: 'ChangeProfilePic' }))
	})
})

describe("text", () => {
	// beforeEach(() => {
	//     jest.setTimeout(10000);
	// });

	// beforeAll((done /* Call it or remove it */) => {
	//     done(); // Calling it
	// });

	afterAll(() => {
		jest.clearAllTimers()
	})

	it("render loading state", () => {
		const component = renderer.create(
			<MockedProvider>
				<HomeScreen {...mockedParams} />
			</MockedProvider>
		);

		const name = component.root.findByProps({ testID: 'nameDisplay' }).props
		expect(name.children).toEqual('loading');

		const bio = component.root.findByProps({ testID: 'bioDisplay' }).props
		expect(bio.children).toEqual('loading');
		// done()
	});

	it("render data state", async () => {
		const mock = [
			{
				request: {
					query: USER_DETAILS,
					variables: {
						id: 1
					}
				},
				result: () => {
					console.log('this ran!')
					return {
						data: {
							getUserDetails: {
								name: "Byung ho",
								bio: "Photographer",
								image_path: "file:///storage/emulated/0/Android/data/com.story/files/Pictures/3d8ffa05-c1cb-4809-9d4c-48b7b81b9516.jpg"
							}
						}
					}
				}
			}
		]

		const component = renderer.create(
			<MockedProvider
				mocks={mock}
				addTypename={false}
			>
				<HomeScreen {...mockedParams} />
			</MockedProvider>
		);

		console.log('here', component.toJSON())

		// await wait(100);
		await new Promise(resolve => setTimeout(resolve, 100));

		console.log('not here', component.toJSON())



		const name = component.root.findByProps({ testID: 'nameDisplay' }).props
		expect(name.children).toEqual('Byung ho');

		const bio = component.root.findByProps({ testID: 'bioDisplay' }).props
		expect(bio.children).toEqual('Photographer');

	})

	it('render error state', async () => {
		const mock = [
			{
				request: {
					query: USER_DETAILS,
					variables: { id: 1 }
				},
				error: new Error('erroe'),
			}
		]

		const component = renderer.create(
			<MockedProvider mocks={mock} addTypename={false}>
				<HomeScreen {...mockedParams} />
			</MockedProvider>
		);

		await wait(1000);
		const name = component.root.findByProps({ testID: 'nameDisplay' }).props
		expect(name.children).toEqual('Error');


		const bio = component.root.findByProps({ testID: 'bioDisplay' }).props
		expect(bio.children).toEqual('Error');

	})
})
