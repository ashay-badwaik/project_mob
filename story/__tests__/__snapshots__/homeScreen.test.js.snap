// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "height": 100,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 350,
        "justifyContent": "center",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="400"
      bbWidth="400"
      focusable={false}
      height="400"
      meetOrSlice={0}
      minX={-800}
      minY={-800}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "justifyContent": "center",
            "position": "absolute",
          },
          Object {
            "flex": 0,
            "height": 400,
            "width": 400,
          },
        ]
      }
      vbHeight={1600}
      vbWidth={1600}
      width="400"
    >
      <RNSVGGroup>
        <RNSVGPath
          d="m754,0 L610,443.19 L233,717.1 L-233,717.1       -610,443.19 -754,0 -610,-443.19       -233,-717.1 233,-717.1       610,-443.19       z"
          fill={null}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={0}
          strokeWidth="30"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNCMaskedView>
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="300"
          bbWidth="300"
          focusable={false}
          height="300"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 300,
                "width": 300,
              },
            ]
          }
          vbHeight={280}
          vbWidth={300}
          width="300"
        >
          <RNSVGGroup>
            <RNSVGGroup
              fill={4279918340}
              matrix={
                Array [
                  0.1,
                  0,
                  0,
                  -0.1,
                  0,
                  280,
                ]
              }
              propList={
                Array [
                  "fill",
                  "stroke",
                ]
              }
              stroke={null}
            >
              <RNSVGPath
                d="M1011 2703 c-94 -68 -255 -185 -318 -232 -85 -63 -195 -143 -270 -196 l-61 -44 -52 -163 c-29 -90 -60 -185 -71 -212 -10 -27 -19 -55 -19 -62 0 -8 -4 -22 -9 -32 -5 -9 -34 -94 -64 -189 l-54 -172 54 -168 c30 -92 58 -177 62 -188 5 -11 22 -63 39 -115 111 -353 115 -361 141 -378 14 -9 56 -39 93 -67 37 -27 95 -70 129 -95 35 -25 85 -61 111 -80 26 -19 81 -59 120 -88 40 -28 107 -77 148 -107 l75 -55 436 0 435 0 55 40 c30 21 69 49 87 62 18 13 54 39 80 58 26 20 79 58 117 85 39 27 75 54 81 60 6 5 46 35 88 65 43 30 105 75 138 100 68 51 43 -8 207 500 46 140 92 281 102 312 20 58 20 58 1 115 -11 32 -41 126 -67 208 -26 83 -51 159 -55 170 -5 11 -35 103 -67 205 l-59 185 -50 38 c-27 21 -83 62 -124 92 -41 30 -133 98 -205 150 -131 97 -146 108 -265 190 l-65 45 -436 0 -436 0 -52 -37z"
                propList={
                  Array [
                    "strokeWidth",
                  ]
                }
                strokeWidth="20"
              />
            </RNSVGGroup>
            <RNSVGGroup
              fill={4294967295}
              matrix={
                Array [
                  0.1,
                  0,
                  0,
                  -0.1,
                  0,
                  280,
                ]
              }
              propList={
                Array [
                  "fill",
                  "stroke",
                ]
              }
              stroke={null}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "height": 300,
            "justifyContent": "center",
            "opacity": 1,
            "width": 300,
          }
        }
        testID="image"
      >
        <Image
          source={
            Object {
              "testUri": "../../../images/no_profile_pic.jpeg",
            }
          }
          style={
            Object {
              "alignContent": "center",
              "height": 280,
              "width": 280,
            }
          }
        />
      </View>
    </RNCMaskedView>
    <View
      style={
        Object {
          "position": "absolute",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fdbb21",
            "borderRadius": 20,
            "fontSize": 30,
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 90,
            "marginTop": 55,
            "opacity": 1,
            "position": "absolute",
            "width": 40,
          }
        }
        testID="addBtn"
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontSize": 50,
              "lineHeight": 50,
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 300,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#898989",
          "fontSize": 40,
          "textAlign": "center",
        }
      }
      testID="nameDisplay"
    >
      loading
    </Text>
    <Text
      style={
        Object {
          "color": "#898989",
          "fontSize": 30,
          "textAlign": "center",
        }
      }
      testID="bioDisplay"
    >
      loading
    </Text>
  </View>
</RCTSafeAreaView>
`;
